test/common/runtime/runtime_impl_test.cc

```diff
@@ -541,6 +545,29 @@ TEST_F(StaticLoaderImplTest, All) {
   testNewOverrides(*loader_, store_);
 }
 
+#ifdef ENVOY_ENABLE_QUIC
+TEST_F(StaticLoaderImplTest, QuicheReloadableFlags) {
+  // Test that Quiche flags can be overwritten via Envoy runtime config.
+  base_ = TestUtility::parseYaml<ProtobufWkt::Struct>(R"EOF(
+    envoy.reloadable_features.FLAGS_quic_reloadable_flag_quic_testonly_default_false: true
+    envoy.reloadable_features.FLAGS_quic_reloadable_flag_quic_testonly_default_true: false
+  )EOF");
+  setup();
+  EXPECT_EQ(true, GetQuicReloadableFlag(quic_testonly_default_false));
+  EXPECT_EQ(false, GetQuicReloadableFlag(quic_testonly_default_true));
```

Comment:
It may be worth testing how reloadable flags interact with SetQuicReloadableFlag

I think some of the behaviors you'll see:
- Difference between value set by SetQuicReloadableFlag and the flag default.  When no reloadable feature is applied, the SetQuicReloadableFlag value should be returned.
- SetQuicReloadableFlag has no effect if a runtime feature is overriding the value of the reloadable flag.
- When the runtime feature is removed, flag goes back to the value set by SetQuicReloadableFlag

Reply:
Good point. Tests added.
Though in the long run, we are planning to modify how Quiche exports to make sure Envoy gets desired flag values by default.